_posts/2023-12-14-configure-ldap-authentication.md

```diff
@@ -0,0 +1,248 @@
+            password: secret
+            # Base DN to search for users
+            userbase: 'ou=people,dc=example,dc=com'
+            # Filter to search for users (currently in the whole subtree beneath userbase)
+            # {0} is substituted with the username
+            usersearch: '(uid={0})'
+            # Use this attribute from the user as username (if not set then DN is used)
+            username_attribute: cn
+```
+
+Refer to the [Active Directory and LDAP](https://opensearch.org/docs/latest/security/authentication-backends/ldap/) page of the Security plugin for details about the configuration.
+Here we connect to a directory using STARTTLS, bind as a specific user with a password, and find users by their *uid* and use their *cn* attribute as username.
+
+After updating the configuration with `securityadmin.sh` (see above), you should be able to sign-in with any user of the LDAP directory.
+
+For now, once signed-in, almost any action will result in a *permission denied* error because we have not configured permissions yet.
+If you cannot sign-in, check the OpenSearch logs for errors, update the configuration and retry.
+There is no point in continuing if you cannot sign-in at this stage.
+
+## Inspect current permissions
+
+When signed-in, you can check your roles and backend roles by clicking on your avatar in the top-right corner, and clicking on *View roles and identities*.
+A pop-up display *roles* and *backend roles*.
+
+![Screenshot showing the "View roles and identities" link](/assets/media/blog-images/2023-12-14-configure-ldap-authentication/view_roles_and_identities.png){:class="img-centered"}
+
+![Screenshot of the pop-up displaying roles (own\_index) and backend roles (none)](/assets/media/blog-images/2023-12-14-configure-ldap-authentication/roles_backend_roles.png){:class="img-centered"}
+
+*Roles* give actual permissions.
+These permissions are managed using the Security plugin under Management, Security, Roles.
+There, *roles* are also mapped to *backend roles*, that is users belonging to a *backend role* belong to the mapped *role* and have the associated permissions.
+
+*Backend roles* correspond to LDAP groups.
+
+For now, we have not configured authorization in the Security plugin, as a consequence the user does not belong to any *backend role* and only belong to a single role: *own_index* which is automatically added to all users in the default configuration.
+
+## Link roles and backend roles
+
+We have two groups in our directory, *users*, and *admins*.
+All users are members of the *users* group, and administrators are also members of the *admins* group.
+For this simple example, we want all authenticated users to have a read-only access to everything, and administrators to have a read-write access to everything.
+
+In order to do this, we must map the *users* backend role to the *readall* and *kibana\_users* roles, and the *admins* backend role to the *all_access* role.
```

Comment:
I really lack a roles and permissions cheat-sheet… I *feel* like this is correct, but maybe there is more sensible roles to use, in which case I will be happy to learn and adjust the blog post.